styleguide/source/assets/js/nav-primary.js

```diff
@@ -34,99 +34,99 @@ jQuery.noConflict();
-          } else {
-            $('.nav-primary-menu_overlay-mobile').removeClass('nav-primary-menu_overlay-mobile-on');
-          }
-        }, 50);
-      }
+  function removeActive() {
+    setTimeout(function () {
+      $('.nav-primary_section').removeClass('nav-primary_section-active').removeClass('is-hidden');
+    }, 500);
+  }
+
+  function closeOverlay() {
+    $('.nav-primary_overlay-mobile').removeClass('nav-primary_overlay-mobile-on');
+  }
 
+  function changeActive(section) {
+    if (section.hasClass('nav-primary_button')) {
+      // toggle a clicked state on the trigger
+      $(this).toggleClass('nav-primary_button-clicked');
+      // show the overlay
+      showOverlay();
+    } else {
+      section.addClass('nav-primary_section-active');
+      section.siblings('.nav-primary_section').removeClass('nav-primary_section-active');
       if ( $(window).width() < 740 ) {
-        // hide the other primary items
-        $(this).parents('.nav-primary_list-item').siblings('.nav-primary_list-item').toggleClass('is-hidden');
-        $(this).parents('.nav-primary_list-item').removeClass('is-hidden');
-
-        // When you click the back button revert all that stuff above.
-        $(this).siblings('.nav-primary_list_subnav').children('.nav-primary_list_subnav_list-item-back').on('touch click', function () {
-          $('.nav-primary_list-item_title').removeClass('is-active');
-          $('.nav-primary_list-item').removeClass('is-active');
-          $('.nav-primary_list_subnav').removeClass('is-open');
-          $('.nav-primary_list-item').removeClass('is-hidden');
-        });
+        section.siblings('.nav-primary_section').addClass('is-hidden');
       }
-    });
-  });
+    }
+  }
 
-  // Close the menu and sub menu when user taps outside of subnav.
-  $('.body, header:not(ul.nav-primary_list)').on('touchstart click', function (event) {
-    if($('.nav-primary_list_subnav.is-open').length || $('.nav-primary_list.nav-primary_list-open').length) {
-      if (!$(event.target).closest('#nav-primary, .nav-primary-menu_button').length || $(event.target).hasClass('nav-primary_list-item') || $(event.target).hasClass('nav-primary_list_subnav')) {
-        $('.nav-primary_list').removeClass('nav-primary_list-open').addClass('nav-primary_list-closed');
-        $('.nav-primary-menu_overlay-mobile').removeClass('nav-primary-menu_overlay-mobile-on');
-        $('.nav-primary-menu_button').removeClass('nav-primary-menu_button-clicked');
-        $('.nav-primary_list_subnav').removeClass('is-open');
-        $('.nav-primary_list-item_title').removeClass('is-active');
-        $('.nav-primary_list-item').removeClass('is-active');
+  // Handling for click events. When someone clicks the nav, the mobile nav button, or anywhere
+  // on the page if the menu is already open:
+
+  $(document).on('click', '.body-nav-primary-open, .nav-primary_section, .nav-primary_button', function(e) {
```

Comment:
It turns out my bullet 2 above regarding lost functionality answers my bullet 1 here about touch events. The answer is, "yes, you need a `touchstart` event here." I believe touch devices will treat clicking on things like anchors as click events as weel as touch events, so click listeners capture them, but in this case just touching anywhere on the `.body-nav-primary-open` class doesn't seem to trigger the event.